lib/faraday_middleware/response/caching.rb

```diff
@@ -79,9 +79,13 @@ def cache_on_complete(env)
         finalize_response(cached_response, env)
       else
         response = @app.call(env)
-        if CACHEABLE_STATUS_CODES.include?(response.status)
-          response.on_complete { cache.write(key, response) }
+        # response.status is nil at this point, any checks need to be done inside on_complete block
+        response.on_complete do
```

Comment:
I would prefer you to use

```ruby
 @app.call(env).on_complete do |response|
```

instead of the longer

```ruby
response = @app.call(env)
response.on_complete do
```

I also think is better to use the block parameter rather than the scope `response`